File: javascript/space-age/space-age.js
var SpaceAge = function(seconds){
  this.seconds = seconds;
};

SpaceAge.prototype.onEarth = function(){
  return Number((this.seconds / 31557600).toFixed(2));
};

SpaceAge.prototype.onMercury = function(){
  return this.yearCalculator(4.1519);
};

SpaceAge.prototype.onVenus = function(){
  return this.yearCalculator(1.624);
};

SpaceAge.prototype.onMars = function(){
  return this.yearCalculator(0.5316);
};

SpaceAge.prototype.onJupiter = function(){
  return this.yearCalculator(0.0843);
};

SpaceAge.prototype.onSaturn = function(){
  return this.yearCalculator(0.0340);
};

SpaceAge.prototype.onUranus = function(){
  return this.yearCalculator(0.0118953);
};

SpaceAge.prototype.onNeptune = function(){
  return this.yearCalculator(0.00607);
};

SpaceAge.prototype.yearCalculator = function(delta){
  return Number((this.onEarth() * delta).toFixed(2));
};

module.exports = SpaceAge;

File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    double seconds;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double getSeconds() {
        return seconds;
    }

    double onEarth() {
        return getSeconds() / 31556791;
    }

    double onMercury() {
        return 4.1518 * onEarth();
    }

    double onVenus() {
        return 1.62458 * onEarth();
    }

    double onMars() {
        return 0.53162 * onEarth();
    }

    double onJupiter() {
        return 0.08432 * onEarth();
    }

    double onSaturn() {
        return 0.03397 * onEarth();
    }

    double onUranus() {
        return 0.01189 * onEarth();
    }

    double onNeptune() {
        return 0.006073 * onEarth();
    }

}
